spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -163,46 +156,54 @@ private StartJobRequest getStartJobRequestForIndexRequest(
     StartJobRequest startJobRequest;
     IndexDetails indexDetails = SQLQueryUtils.extractIndexDetails(dispatchQueryRequest.getQuery());
     FullyQualifiedTableName fullyQualifiedTableName = indexDetails.getFullyQualifiedTableName();
-    if (fullyQualifiedTableName.getDatasourceName() == null) {
-      throw new UnsupportedOperationException("Queries without a datasource are not supported");
-    }
     dataSourceUserAuthorizationHelper.authorizeDataSource(
-        this.dataSourceService.getRawDataSourceMetadata(
-            fullyQualifiedTableName.getDatasourceName()));
+        this.dataSourceService.getRawDataSourceMetadata(dispatchQueryRequest.getDatasource()));
     String jobName =
         getJobNameForIndexQuery(dispatchQueryRequest, indexDetails, fullyQualifiedTableName);
-    Map<String, String> tags =
-        getDefaultTagsForJobSubmission(dispatchQueryRequest, fullyQualifiedTableName);
+    Map<String, String> tags = getDefaultTagsForJobSubmission(dispatchQueryRequest);
     tags.put(INDEX_TAG_KEY, indexDetails.getIndexName());
+    tags.put(TABLE_TAG_KEY, fullyQualifiedTableName.getTableName());
+    tags.put(SCHEMA_TAG_KEY, fullyQualifiedTableName.getSchemaName());
     startJobRequest =
         new StartJobRequest(
             dispatchQueryRequest.getQuery(),
             jobName,
             dispatchQueryRequest.getApplicationId(),
             dispatchQueryRequest.getExecutionRoleARN(),
-            constructSparkParameters(fullyQualifiedTableName.getDatasourceName()),
+            constructSparkParameters(dispatchQueryRequest.getDatasource()),
             tags);
     return startJobRequest;
   }
 
   private static Map<String, String> getDefaultTagsForJobSubmission(
-      DispatchQueryRequest dispatchQueryRequest, FullyQualifiedTableName fullyQualifiedTableName) {
+      DispatchQueryRequest dispatchQueryRequest) {
     Map<String, String> tags = new HashMap<>();
     tags.put(CLUSTER_NAME_TAG_KEY, dispatchQueryRequest.getClusterName());
-    tags.put(DATASOURCE_TAG_KEY, fullyQualifiedTableName.getDatasourceName());
-    tags.put(SCHEMA_TAG_KEY, fullyQualifiedTableName.getSchemaName());
-    tags.put(TABLE_TAG_KEY, fullyQualifiedTableName.getTableName());
+    tags.put(DATASOURCE_TAG_KEY, dispatchQueryRequest.getDatasource());
     return tags;
   }
 
+  // Our queries work with datasource name and without datasource name.
+  // Inorder to have a constant jobName in both the scenarios,
+  // we are adding data source name from dispatcher request to the jobName.
   private static String getJobNameForIndexQuery(
       DispatchQueryRequest dispatchQueryRequest,
       IndexDetails indexDetails,
       FullyQualifiedTableName fullyQualifiedTableName) {
-    return dispatchQueryRequest.getClusterName()
-        + ":"
-        + fullyQualifiedTableName.getFullyQualifiedName()
-        + "."
-        + indexDetails.getIndexName();
+    if (fullyQualifiedTableName.getDatasourceName() == null) {
```

Comment:
datasource always exist, right? we don't need to parse from fullyQualifiedTableName.

Reply:
JobName for Index Query is `ClusterName:fullyQualifiedTableName.IndexName.`
So we depend on `fullyQualifiedTableName.getFullyQualifiedName()` for jobName Creation.
Below are the two usecases for if clause.
```
{
"datasource" : "my_glue",
"lang" : "SQL",
"query" : "select * from my_glue.default.logs"
}
```

```
No datasource in query.
{
"datasource" : "my_glue",
"lang" : "SQL",
"query" : "select * from.default.logs"
}
```